docs/history/evolution-of-design-time-assemblies.md

```diff
@@ -0,0 +1,53 @@
+- **Reference Assembly:** This is a standard IL assembly but it only contains metadata and no IL code. In other words only the declarations and no definitions, you can think of this similar to a C++ header file.
+- **Contract Assembly:** This is a reference assembly which has been designed to contain a specific set of APIs only. Before .NET Core the APIs in a reference assembly matched 1-to-1 with the corresponding implementation assembly. Contracts are by design not that way, instead we pick and choose the APIs we want to expose and then make the implementation support them. Moreover, in the past reference assemblies had an assembly identity matching that of the corresponding implementation assembly. Contracts break away from that relationship. Simply observing a Contract Assembly does not reveal the relation to the implementation assembly/-ies that may be available at runtime.
+- **Facade Assembly:** This is a standard IL assembly which contains only type forwards. Since type forwards are stored in the exported type table this assembly doesn’t contain any IL and minimal metadata (only the assembly level metadata).
+- **Partial Facade Assembly:** This is similar to a Facade Assembly but it contains a mixture of type forwards as well as some type definitions.
+- **Design time Assembly:** An assembly that is used by tools on a developer machine. Such tools typically include compilers (C#, VB, as well as XAML) and various msbuild and VS tools. They generally don’t need to contain any implementation because they are primarily an abstraction for what a given platform supports.
+- **Runtime Assembly:** This is the raw assembly used by the runtime on an end user’s machine. These are consumed by the runtime (aka CLR/CoreCLR) and need actual code in them to execute.
+
+## Evolution of Design time assemblies
+
+### Using implementation assemblies
+In the beginning, when building .NET applications, people directly compiled against the implementation assemblies that existed on the developer’s machine, so the design time and runtime assemblies were exactly the same.
+
+![](implementation-assemblies.png)
+<BR/> *Figure 1 - Implementation Assemblies*
+
+### Using reference assemblies
+One of the big issues with taking direct dependencies on the implementation assemblies is that the developer is tightly coupled to a particular framework version. In a world where we have multiple versions of the .Net framework developers need to be able to write an application that can work on any particular version. To support this scenario we created a feature called multi-targeting in which we created a set of reference assemblies for each version of the framework. We referred to these sets of reference assemblies as Targeting Packs. Targeting Packs are only deployed to the developers machine and used by tools to ensure developers didn’t use APIs that weren’t supported on the particular version they were targeting.
```

Comment:
In the sentence with `multiple versions of the .Net framework` -> capitalize to `.NET Framework` and maybe add `(v4.6, v4.6.1, v4.6.2, ...)` as an example